rero_mef/monitoring/api.py

```diff
@@ -0,0 +1,258 @@
+                        if missing_in_es:
+                            info[doc_type]['es-'] = list(missing_in_es)
+        return info
+
+    @classmethod
+    def check(cls, with_deleted=False, difference_db_es=False):
+        """Compaire elasticsearch with database counts.
+
+        :param with_deleted: count also deleted items in database.
+        :return: dictionair with all document types with a difference in
+        database and elasticsearch counts.
+        """
+        checks = {}
+        for info, data in cls.info(
+            with_deleted=with_deleted,
+            difference_db_es=difference_db_es
+        ).items():
+            db_es = data.get('db-es', '')
+            if db_es not in [0, '']:
+                checks.setdefault(info, {})
+                checks[info]['db_es'] = db_es
+            if data.get('db-'):
+                checks.setdefault(info, {})
+                checks[info]['db-'] = len(data.get('db-'))
+            if data.get('es-'):
+                checks.setdefault(info, {})
+                checks[info]['es-'] = len(data.get('es-'))
+        return checks
+
+    @classmethod
+    def check_mef(cls):
+        """Compaire MEF and entities counts.
+
+        returns: MEF, entities and MEF-entities counts.
+        """
+        checks = {}
+        for mef in get_mefs_endpoints():
+            mef_search = mef['mef_class'].search
+            for entity in mef['endpoints']:
+                entity_class = get_entity_class(entity)
+                mef_count = mef_search() \
+                    .filter('exists', field=entity_class.name) \
+                    .count()
+                db_count = entity_class.count()
+                checks[entity] = {
+                    'mef': mef_count,
+                    'db': db_count,
+                    'mef-db': mef_count - db_count
+                }
+        return checks
+
+    @classmethod
+    def missing(cls, doc_type, with_deleted=False):
+        """Get missing pids.
+
+        Get missing pids in database and elasticsearch and find duplicate
+        pids in elasticsearch.
+
+        :param doc_type: doc type to get missing pids.
+        :return: dictionair with all missing pids.
+        """
+        missing_in_db, missing_in_es, pids_es_double, index =\
+            cls.get_es_db_missing_pids(
+                doc_type=doc_type,
+                with_deleted=with_deleted
+            )
+        if index:
+            return {
+                'DB': list(missing_in_db),
+                'ES': list(missing_in_es),
+                'ES duplicate': pids_es_double
+            }
+        else:
+            return {'ERROR': 'Document type not found: {doc_type}'}
+
+    @classmethod
+    def print_missing(cls, doc_type):
+        """Print missing pids for the given document type.
+
+        :param doc_type: doc type to print.
+        """
+        for info, data in cls.missing(doc_type=doc_type).items():
```

Comment:
I do not think, the two `else` are what you want. I think you can simplify this with `elif`